Block/NodeType/AbstractNode.php

```diff
@@ -125,4 +130,15 @@ public function isViewAllLinkAllowed()
     {
         return $this->viewAllLink;
     }
+
+    public function _toHtml()
+    {
+        $template = $this->templateResolver->getMenuTemplate(
+            $this->getMenuCode(),
+            $this->_template
+        );
+        $this->setTemplate($template);
```

Comment:
can we set the template in the class constructor?

Reply:
In constructor we don't have access to menu code variable